File: C++/代码随想录/Hashtable/349. Intersection of Two Arrays.cpp
// 349. Intersection of Two Arrays
// https://leetcode.cn/problems/intersection-of-two-arrays/

#include <iostream>
#include <vector>
#include <unordered_set>

std::vector<int> intersection(std::vector<int>& nums1, std::vector<int>& nums2) {
    std::unordered_set<int> set1(nums1.begin(), nums1.end());
    std::unordered_set<int> resultSet;

    for (int num : nums2) {
        if (set1.count(num)) {
            resultSet.insert(num);
        }
    }

    return std::vector<int>(resultSet.begin(), resultSet.end());
}


int main() {
    // std::vector<int> nums1 = {1, 2, 2, 1};
    // std::vector<int> nums2 = {2, 2};
    std::vector<int> nums1 = {4,9,5};
    std::vector<int> nums2 = {9,4,9,8,4};
    std::vector<int> res = intersection(nums1, nums2);
    for (int i : res) {
        std::cout << i << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: C++/代码随想录/Hashtable/349. Intersection of Two Arrays.py
from typing import List

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1, set2 = set(nums1), set(nums2)
        return list(set1 & set2)

sol = Solution()

# 测试用例
print(sol.intersection([1, 2, 2, 1], [2, 2]))  # 应该输出 [2]
print(sol.intersection([4, 9, 5], [9, 4, 9, 8, 4]))  # 应该输出 [9, 4]